src/Endemic/Configuration/Types.hs

```diff
@@ -279,8 +282,18 @@ data CompileConfig = CompConf
     modBase :: [FilePath],
     -- | Any additional targets to include
     additionalTargets :: [FilePath],
-    -- | the "depth" of the holes, see general notes on this
-    hole_lvl :: Int,
+    -- | The "level" and "depth" of the holes, see general notes on this.
+    -- holeLvl determines how many additional holes a hole fit can have.
+    -- Note: a holeLvl > 0 means that we must have a holeDepth of > 0,
+    -- since we *must* fill all holes.
+    holeLvl :: Int,
+    -- |  Hole depth determines how many levels of additional holes we allow.
+    -- I.e. we have holeLvl = 1 then and x :: a, y :: a -> a in scope, a hole
+    -- _ :: a would give the following:
```

Comment:
Uh, this example ... I do not fully understand it. 

So ... Hole-Level says how many holes can be in an expression,  
and Hole-Depth says that a hole can be filled with a hole up to depth n ?

Reply:
Yes, exactly!